### DISCUSSION TITLE
A question about empty line

### DISCUSSION THREAD

**Discussion by KuiyueRO** [ycnmhd/obsidian-lineage#25] [Q&A]
I noticed that there is code like this in `main.js`, I would like to ask why empty lines should be replaced with `&nbsp`·.

thanks

```javascript
// src/view/actions/markdown-preview-action.ts
var markdownPreviewAction = (element2, nodeId) => {
  const plugin = getPlugin();
  const view = getView();
  const store = view.documentStore;
  const render = (content) => {
    if (view && element2) {
      element2.empty();
      if (content.length > 0 && !/^> /.test(content)) {
        content = content.replace(/^$/gm, "&nbsp;");
      }
      import_obsidian3.MarkdownRenderer.render(
        plugin.app,
        content,
        element2,
        store.getValue().file.path,
        view
      );
    }
  };
  const $content = contentStore(view, nodeId);
  const unsub = $content.subscribe((content) => {
    render(content);
  });
  return {
    destroy: () => {
      unsub();
    }
  };
};
```

**Comment by ycnmhd** [comment]
It prevents [Markdown Renderer](https://docs.obsidian.md/Reference/TypeScript+API/MarkdownRenderer/render) from collapsing empty lines.

With `content = content.replace(/^$/gm, "&nbsp;");`
![2024-05-25_21-08-25](https://github.com/ycnmhd/obsidian-lineage/assets/33285087/4b1d6a9d-0c57-4368-b3f8-318bb6bdf622)

Without `content = content.replace(/^$/gm, "&nbsp;");`
![2024-05-25_21-07-09](https://github.com/ycnmhd/obsidian-lineage/assets/33285087/fc339ad1-71d5-45c4-bb3f-8a22588acd00)

Edit: This code doesn't alter the saved file. It only affects how text is displayed.
